presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedScalarSubquery.java

```diff
@@ -14,6 +14,7 @@
 package io.prestosql.sql.planner.iterative.rule;
```

Comment:
Can you please add some tests for INNER vs LEFT?
Do you have any query that had incorrect result?

Reply:
This is implicitly covered by existing tests. Writing direct tests is not possible because it is not possible to create LEFT lateral join by hand.